cookbooks/continuous_delivery.rst

```diff
@@ -0,0 +1,187 @@
+* `Creating and deleting environments <cd-env-create-delete>`_
+* `URLs <cd-url>`_
+
+.. _cd-requirements:
+
+Requirements
+^^^^^^^^^^^^
+
+The systement implementing a custom SymfonyCloud deployment automation will have
+several requirements:
+
+#. Have the `Symfony CLI <https://symfony.com/download>`_ installed and runnable
+#. Have Git installed and runnable #. Have `authentication`_ completed
+
+Authentication
+^^^^^^^^^^^^^^
+
+If the automation is expected to be run interactivtely by humans, thoses humans
+must first - locally on their machine - setup Git and run ``symfony login`` as
+well as ``symfony account:ssh:key:add``.
+
+.. _machine-user:
+
+In the implementation is to be run in a non-interactive environment (in a system
+like Jenkins, Travis, GitLab CI or GitHub actions for example), we recommend the
+use of a "Machine user". To do so, head to `SymfonyConnect
+<https://connect.symfony.com>`_ and register a new account.
+
+As this context will not be interactive, interactive authentication will not be
+possible, this is why you have to use an :doc:`API token
+</cookbooks/api_tokens>`, additionnaly, as deployment are based on Git, you will
+have to generate an SSH key, and register it in SymfonyCloud:
+
+.. code-block:: bash
+
+   ssh-keygen -N '' -f symfonycloud-ci.key SYMFONY_TOKEN=xxxx symfony
+   account:ssh:key:add symfonycloud-ci.key
+
+.. caution::
+
+   Don't forget to register and expose the ``SYMFONY_TOKEN`` environment value
+   as well as the generated SSH key in your automation system.
+
+.. _cd-env-flags:
+
+CLI flags and environment variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The CLI tool defines several flags and environment flags that allow to change
+it's behavior. As a general rule, for automated context, environment variables
+are preferable but flags can be used ponctually if this is required.
+
+SYMFONY_PROJECT
+"""""""""""""""
+
+Setting this environment value make useless to run ``symfony project:link``. Its
+value should be set to the SymfonyCloud project's id that can be found with
+``symfony project:info`` or in the emails sent by SymfonyCloud.
+
+.. tip:: One can also use the ``--project`` flag.
+
+SYMFONY_ENVIRONMENT
+"""""""""""""""""""
+
+By default, when SymfonyCloud needs to work on the remote environment, it tries
+to find an exising one whose name matches the current Git branch name. One might
+want to change this logic, this can be done by setting this environment variable
+with a different value. You can use the ``symfony envs`` commands to list
+environments currently present on SymfonyCloud.
+
+.. tip:: One can also use the ``--environment`` flag.
+
+.. _cd-env-create-delete:
+
+Creating and deleting environments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``symfony env:create <name> <parent>`` command creates a new SymfonyCloud
+environment. Most users don't need to provide any argument when running it
+locally, but when running inside a script it is best to specify them. The first
```

Comment:
This sentence appears unclear.

What does _most users_ mean? 

Perhaps it would be better to clarify that there are

- optional arguments that default to specific values
- options that default to specific values

rather than vaguely suggesting that a reader could belong to the group of _most users_.

What  do you think?